os/mm/mm_heap/mm_free.c

```diff
@@ -73,6 +73,22 @@
 /****************************************************************************
  * Private Functions
  ****************************************************************************/
+static void mm_add_delaylist(FAR struct mm_heap_s *heap, FAR void *mem)
```

Comment:
Yes, it is different. I understand, sched_free is doing a free() on lpwork thread, so It is not ISR. 

But time_release and wd_delete using sched_free is called very often on ISR.
So I think, It takes a long time for these to actually become free because lpwork is low priority.
This will occupy unnecessary memory.

How do you think about sched_free?
We would like to change the delay free() on lpwork.

Reply:
Agreed. 
Let us discuss about this and take a decision. 
However, since this is not related to SMP, let us not keep this as part of this PR. 
Maybe you can raise an issue in github so that we can track it.